server/handler/assignments.go

```diff
@@ -0,0 +1,72 @@
+package handler
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/src-d/code-annotation/server/middleware"
+	"github.com/src-d/code-annotation/server/repository"
+	"github.com/src-d/code-annotation/server/serializer"
+
+	"github.com/go-chi/chi"
+)
+
+func GetAssignmentsForUserExperiment(w http.ResponseWriter, r *http.Request) {
+	//TODO: !ok
+	postponed, _ := w.(middleware.PostponedResponse)
```

Comment:
what is the point to "postpone" response? I checked handlers and I don't see where you might need it. Every time you `return` you can create normal response from the line just above that return.

Reply:
the postponed Responses are the way to avoid writing things randomly; that way the handlers push the results to the postponed Response, and at the end of the flow it is processed and properly serialized, the errors are checked...

Reply:
you are talking about transformation. Not postponing.

Postponing is something like
```go
func handler(w) {
  writeSomethingTo(w)
  // do long computation maybe in goroutine
  writeMoreTo(w)
}
```

Reply:
The Response is written after the handler is run

Reply:
I get it. I'm saying the name is confusing. If you want to keep using this approach (I don't) the name like `Responser` is much better. Because it's interface and consumer doesn't need to know how it works internally (does it postpone anything or not)

Reply:
functionality dropped